fdm-data/src/fertilizers/hash.ts

```
@@ -0,0 +1,26 @@
+    fertilizer.hash = null
+
+    // Remove all keys without a value
+    const filteredFertilizer = Object.fromEntries(
+        Object.entries(fertilizer).filter(
+            ([, value]) => value !== undefined || value !== null,
+        ),
+    )
+
+    // Sort keys to ensure consistent hash generation for identical objects
+    const sortedKeys = Object.keys(filteredFertilizer).sort()
+    const sortedFertilizer = sortedKeys.reduce((obj: any, key) => {
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)